src/Symfony/Component/HttpClient/Response/MockResponse.php

```diff
@@ -121,6 +139,8 @@ public static function fromRequest(string $method, string $url, array $options,
 
         if ($mock instanceof self) {
             $mock->requestOptions = $response->requestOptions;
+            $mock->requestMethod = $method;
+            $mock->requestUrl = $url;
```

Comment:
> Would it make sense to do `$mock->info = &$response->info;` instead? That would allow checking any option from the response when having the mock at hand, via `getInfo()`. Can you give it a try?

```suggestion
            $mock->requestMethod = $method;
            $mock->requestUrl = $url;
            $mock->info = &$response->info;
```

@nicolas-grekas thanks for the response! I've tried your suggestion and it worked. It also passes the unit test of `MockResponseTest` asserting:

```php
        $this->assertSame($method, $responseMock->getInfo('http_method'));
        $this->assertSame($url, $responseMock->getInfo('url'));
``` 

But there are another 5 errors and 11 failures unit tests running `$ php ./phpunit src/Symfony/Component/HttpClient/` folder. Before creating these new two getter methods I've also tried

```suggestion
            $mock->info['http_method`] = $method;
            $mock->info['url`] = $url;
```

And I also have issues with the unit tests asserting the `url` with some test of redirects. Should I try in another way?

Reply:
no need, that was a bad idea :)